packages/application-shell-connectors/src/components/application-context/application-context.tsx

```diff
@@ -0,0 +1,217 @@
+import React from 'react';
+import moment from 'moment-timezone';
+import getDisplayName from '../../utils/get-display-name';
+
+type AdditionalProperties = { [key: string]: unknown };
+type TRawUser = {
+  id: string;
+  email: string;
+  firstName: string;
+  lastName: string;
+  language: string;
+  timeZone?: string;
+} & AdditionalProperties;
+type TApplicationContextUser = {
+  id: string;
+  email: string;
+  firstName: string;
+  lastName: string;
+  locale: string;
+  timeZone?: string;
+};
+type TApplicationContextPermissions = { [key: string]: boolean };
+type TRawProject = {
+  key: string;
+  version: number;
+  name: string;
+  countries: string[];
+  currencies: string[];
+  languages: string[];
+  owner: {
+    id: string;
+  };
+  permissions: TApplicationContextPermissions;
+} & AdditionalProperties;
+type TApplicationContextProject = {
+  key: string;
+  version: number;
+  name: string;
+  countries: string[];
+  currencies: string[];
+  languages: string[];
+  ownerId: string;
+};
+type TApplicationContextEnvironment = {
+  applicationName?: string;
+  frontendHost: string;
+  mcApiUrl: string;
+  location: string;
+  env: string;
+  cdnUrl: string;
+  servedByProxy: string | boolean;
+};
+type TApplicationContext<ExtendedEnvironment extends {}> = {
+  environment: ExtendedEnvironment & TApplicationContextEnvironment;
+  user: TApplicationContextUser | null;
+  project: TApplicationContextProject | null;
+  permissions: TApplicationContextPermissions | null;
+  dataLocale: string | null;
+};
+type ProviderProps<ExtendedEnvironment extends {}> = {
+  environment: ExtendedEnvironment & TApplicationContextEnvironment;
+  user?: TRawUser;
+  project?: TRawProject;
+  projectDataLocale?: string;
+  children: React.ReactNode;
```

Comment:
I found this article with some dos and donts about TS, and the guy mentioned to explicitly type the children props.

https://medium.com/@martin_hotell/10-typescript-pro-tips-patterns-with-or-without-react-5799488d6680#55bb

Reply:
Yeah. React.Node is the correct type here to what I know.